src/core/HttpResponse.ts

```diff
@@ -11,16 +11,14 @@ export interface HttpResponseInit extends ResponseInit {
 
 declare const bodyType: unique symbol
 
-export interface StrictRequest<BodyType extends DefaultBodyType>
-  extends Request {
+export interface StrictRequest<BodyType extends JsonBodyType> extends Request {
```

Comment:
Technically, `StrictRequest` only annotates the JSON body reading method so it cannot be of any other type than the `JsonBodyType`. As in, it cannot suddenly return a stream.